### PR TITLE
`DerivePConstantViaEnum`

### PR DESCRIPTION
**Pull request by chfanghr**

### CODE REVIEW

File: src/Plutarch/Extra/IsData.hs [hunk 1]
```diff
@@ -156,6 +171,24 @@ instance forall (a :: PType). PIsData (PEnumData a) where
...
+
+-- | @since 1.1.0
+instance (PLift p, Enum h) => PConstantDecl (DerivePConstantViaEnum h p) where
+    type PConstantRepr (DerivePConstantViaEnum h p) = Integer
+    type PConstanted (DerivePConstantViaEnum h p) = p
+
+    pconstantToRepr = toInteger . fromEnum @h . coerce
+    pconstantFromRepr = Just . coerce . toEnum @h . fromInteger
+
+-- instance Enum a => PConstantDecl (EnumIsData a)
```
Comment (emiflake) [inline src/Plutarch/Extra/IsData.hs:190]: Remove this.

**Review by emiflake — APPROVED** [review]
Lovely. :)

File: src/Plutarch/Extra/IsData.hs (hunk 1 above)
Comment (chfanghr) [reply src/Plutarch/Extra/IsData.hs:190]: Oops, I forgot.

**Review by t1lde — APPROVED** [review]
lgtm :+1: 
thanks for this :)

**Pull request merged**